Mathlib/Data/Nat/Defs.lean

```diff
@@ -770,6 +770,10 @@ protected lemma pow_right_injective (ha : 2 ≤ a) : Injective (a ^ ·) :=by
   | 0 => by simp
   | n + 1 => by rw [Nat.pow_succ, mul_eq_zero, Nat.pow_eq_zero]; omega
 
+/-- For `a > 1`, `a ^ b = a` iff `b = 1`. -/
+lemma pow_eq_self_iff (a b : ℕ) (ha : 2 ≤ a) : a ^ b = a ↔ b = 1 :=
```

Comment:
I think `1 < a` might be preferrable to `2 ≤ a` (although they are definitionally equal).

Reply:
Done.